tests/test_protocol.py

```diff
@@ -0,0 +1,63 @@
+"""
+Runs tests that ensure protocol invariants
+
+TODO add other tests including:
+- some CI on postSignatures
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+import avro.schema
+
+import compile_schemas
+
+
+class TestValidateSchemas(unittest.TestCase):
+    """
+    Ensure the schemas conform to certain rules
```

Comment:
We can solve that in a subsequent PR.  I just want to get this framework checked in that has CI run against all subsequent PRs, and then we can argue about what tests we want within that framework and how to structure them.